dpe/src/commands/certify_key.rs

```diff
@@ -329,4 +402,214 @@ mod tests {
+                // note: the cms crate does not let us convert issuer_and_serial_number.issuer
+                // into bytes so we unfortunately cannot test the issuer name here.
+            }
+            _ => panic!("Error: Signer Identifier is not IssuerAndSerialNumber!"),
+        };
+
+        // parse encapsulated content info
+        let econtent_info = &mut signed_data.encap_content_info;
+        assert_eq!(
+            econtent_info.econtent_type.to_string(),
+            "1.2.840.113549.1.7.1"
+        );
+        // skip first 4 explicit encoding bytes
+        let econtent = &econtent_info.econtent.as_mut().unwrap().to_der().unwrap()[4..];
+
+        // validate csr signature with the alias key
+        let csr_digest = env.crypto.hash(DPE_PROFILE.alg_len(), &econtent).unwrap();
+        let priv_key = match DPE_PROFILE.alg_len() {
+            AlgLen::Bit256 => EcKey::private_key_from_der(include_bytes!(
+                "../../../platform/src/test_data/key_256.der"
+            )),
+            AlgLen::Bit384 => EcKey::private_key_from_der(include_bytes!(
+                "../../../platform/src/test_data/key_384.der"
+            )),
+        }
+        .unwrap();
+        let curve = match DPE_PROFILE.alg_len() {
+            AlgLen::Bit256 => Nid::X9_62_PRIME256V1,
+            AlgLen::Bit384 => Nid::SECP384R1,
+        };
+        let group = &EcGroup::from_curve_name(curve).unwrap();
+        let alias_key = EcKey::from_public_key(group, priv_key.public_key()).unwrap();
+        let csr_sig = EcdsaSig::from_der(signer_info.signature.as_bytes()).unwrap();
+        assert!(csr_sig.verify(csr_digest.bytes(), &alias_key).unwrap());
+
+        // validate csr
+        let (_, csr) = X509CertificationRequest::from_der(&econtent).unwrap();
+        let cri = csr.certification_request_info;
+        assert_eq!(cri.version.0, 0);
+        assert_eq!(
+            csr.signature_algorithm.algorithm.to_id_string(),
+            sig_alg_oid
+        );
+
+        // validate certification request info signature
+        EcdsaSig::from_der(csr.signature_value.data.as_ref()).unwrap();
+
+        // validate certification request info subject pki
+        let PublicKey::EC(ec_point) = cri.subject_pki.parsed().unwrap() else {
+            panic!("Error: Failed to parse public key correctly.");
+        };
+        let pub_key_der = ec_point.data();
+        // skip first 0x04 der encoding byte
+        let x = BigNum::from_slice(&pub_key_der[1..DPE_PROFILE.get_ecc_int_size() + 1]).unwrap();
+        let y = BigNum::from_slice(&pub_key_der[DPE_PROFILE.get_ecc_int_size() + 1..]).unwrap();
+        EcKey::from_public_key_affine_coordinates(group, &x, &y).unwrap();
+
+        // note: ideally here, we would verify the signature with the public key.
+        // However, the x509_parser library does not expose a function to
+        // convert the certification request info into bytes so we have no way to
+        // get the digest.
```

Comment:
I think you can use the `raw` field here: https://docs.rs/x509-parser/latest/x509_parser/certification_request/struct.X509CertificationRequestInfo.html

Reply:
Thanks, this worked.